ext/cl_khr_command_buffer.asciidoc

```diff
@@ -0,0 +1,1875 @@
+    const void* pattern,
+    size_t pattern_size,
+    size_t offset,
+    size_t size,
+    cl_uint num_sync_points_in_wait_list,
+    const cl_sync_point_khr* sync_point_wait_list,
+    cl_sync_point_khr* sync_point,
+    cl_mutable_command_khr* mutable_handle);
+
+cl_int clCommandFillImageKHR(
+    cl_command_buffer_khr command_buffer,
+    cl_command_queue command_queue,
+    cl_mem image,
+    const void* fill_color,
+    const size_t* origin,
+    const size_t* region,
+    cl_uint num_sync_points_in_wait_list,
+    const cl_sync_point_khr* sync_point_wait_list,
+    cl_sync_point_khr* sync_point,
+    cl_mutable_command_khr* mutable_handle);
+
+cl_int clCommandNDRangeKernelKHR(
+    cl_command_buffer_khr command_buffer,
+    cl_command_queue command_queue,
+    const cl_ndrange_kernel_command_properties_khr* properties,
+    cl_kernel kernel,
+    cl_uint work_dim,
+    const size_t* global_work_offset,
+    const size_t* global_work_size,
+    const size_t* local_work_size,
+    cl_uint num_sync_points_in_wait_list,
+    const cl_sync_point_khr* sync_point_wait_list,
+    cl_sync_point_khr* sync_point,
+    cl_mutable_command_khr* mutable_handle);
+
+cl_int clGetCommandBufferInfoKHR(
+    cl_command_buffer_khr command_buffer,
+    cl_command_buffer_info_khr param_name,
+    size_t param_value_size,
+    void* param_value,
+    size_t* param_value_size_ret);
+----
+
+
+=== New API Enums
+
+Enums for querying device command-buffer capabilities with
+*clGetDeviceInfo*, see <<command-buffer-queries, device queries table>>:
+
+[source]
+----
+// Accepted values for the param_name parameter to clGetDeviceInfo
+CL_DEVICE_COMMAND_BUFFER_CAPABILITIES_KHR              0x12A9
+CL_DEVICE_COMMAND_BUFFER_REQUIRED_QUEUE_PROPERTIES_KHR 0x12AA
+
+// Bits for cl_device_command_buffer_capabilities_khr bitfield
+CL_COMMAND_BUFFER_CAPABILITY_KERNEL_PRINTF_KHR        (0x1 << 0)
+CL_COMMAND_BUFFER_CAPABILITY_DEVICE_SIDE_ENQUEUE_KHR  (0x1 << 1)
+CL_COMMAND_BUFFER_CAPABILITY_SIMULTANEOUS_USE_KHR     (0x1 << 2)
+CL_COMMAND_BUFFER_CAPABILITY_OUT_OF_ORDER_KHR         (0x1 << 3)
+
+// Values for cl_command_buffer_state_khr
+CL_COMMAND_BUFFER_STATE_RECORDING_KHR              0x0
+CL_COMMAND_BUFFER_STATE_EXECUTABLE_KHR             0x1
+CL_COMMAND_BUFFER_STATE_PENDING_KHR                0x2
+CL_COMMAND_BUFFER_STATE_INVALID_KHR                0x3
+----
+
+Enums for base <<command-buffer, command-buffers>> functionality:
+
+[source]
+----
+// Error codes
+CL_INVALID_COMMAND_BUFFER_KHR                      -1138
+CL_INVALID_SYNC_POINT_WAIT_LIST_KHR                -1139
+CL_INCOMPATIBLE_COMMAND_QUEUE_KHR                  -1140
+
+// Bitfield to clCreateCommandBufferKHR
+CL_COMMAND_BUFFER_PROPERTIES_KHR                   0x1293
+
+// Bits for cl_command_buffer_properties_khr bitfield
+CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR             (0x1 << 0)
+
+// cl_command_buffer_info_khr queries to clGetCommandBufferInfoKHR
+CL_COMMAND_BUFFER_INFO_QUEUES_KHR                  0x1294
```

Comment:
Minor/consistency: Most of the other query enums do not include "INFO" in their names.  For example, we have a query enum `CL_CONTEXT_REFERENCE_COUNT` and not `CL_CONTEXT_INFO_REFERENCE_COUNT`.  Should we drop "INFO" in the names of these enums?

Note that the enum type does typically include "info" in the name, so we should keep `cl_command_buffer_info_khr`.

Reply:
Tagging @EwanC for awareness.

Reply:
Ah yes, good catch Ben. Happy for you to remove the `_INFO_` component of these enums.

Reply:
I removed "INFO" in commit https://github.com/KhronosGroup/OpenCL-Docs/pull/711/commits/929586586711de05bd45c05840c3b3d147d4e949 - please take a look.

Reply:
Looks good, updated my headers PR in https://github.com/KhronosGroup/OpenCL-Headers/pull/182/commits/eba2023490a4071718bad97ffbef6f877f41bb2e to match